podman.md

```diff
@@ -0,0 +1,123 @@
+
+First, `build_vars`. This is the first function that is called in the
+script. It sets up variables that will be used in the script and by
+other code in `container-build`. The most important is
+`build_image_base`. This is set to `radiasoft/python3`. Podman images
+are built in "layers". What that means is an image can be built on top
+of another image. In our case we are building the SlicOps image on top
+of the `radiasoft/python3` image. The `radiasoft/python3` image is a
+Fedora image with some of the basic necessities (e.g. gcc) installed.
+Importantly for us, it has `python3` installed (SlicOps is a Python
+application).
+
+Next up is `build_as_root`. This function does all of its steps as the
+root user. For SlicOps that is installing packages like Node.JS and
+some of the EPICS dependencies.
+
+Finally, we have `build_as_run_user`. Similar to `build_as_root` this
+runs install steps. But, it runs them as the `run_user`. That is the
+user that will be running the application inside of the container. For
+SliCops this step installs EPICS and installs the SlicOps package.
+
+Once all of these steps are run then the image has all of the
+dependencies that are required to run SliCops.
+
+Running `container-build` will take many minutes.
+
+## Running the image
+
+Now we have built (or you got from someone / downloaded) the image and
+we are ready to run it.
+
+To run the image
+
+```
+podman run --rm --interactive --tty --network=host slaclab/slicops /bin/bash
+```
+
+Breaking down the parts of that command:
+- `podman`: Call the podman program
+- `run`: The command we want. In this case, to run the image.
+- `--rm`: Automatically remove the container when it exits. If we
+  don't have this flag the container will be left "paused" when you
+  exit. These paused containers tend to accumulate and just take up
+  space on your machine.
+- `--interactive`: Keep STDIN open. This will allow us to enter
+  commands inside of the running container.
+- `--tt`: Allocate a "pseudo-teletypewriter". Basically, make working
+  inside of the container in a shell work like how a using a terminal
+  works on your computer.
+- `--network=host`: Use the host machines network stack. This means
+  that any http server started in the container will appear as if it
+  was started on your computer. You'll be able to connect to it over
+  localhost.
+- `slaclab/slicops`: This is the name of the image we want to run.
+- `/bin/bash`: Run the bash shell when the container is started
+
+Once the container is started you will be dropped into a bash shell.
+From there you can start the necessary software.
+
+First, start the EPICS sim detector which is a simulated detector that the rest of the application
+will use to read images from.
+```
+slicops epics sim-detector
+```
+
+Next, start the "ui-api". This is the Python web server that
+interfaces between EPICS and the GUI to read from PV's and send the
+outputs to the browser.
+
+```
+slicops service ui-api
```

Comment:
&> ui-api.log &

May want to teach about podman exec

Reply:
d1aa076